src/main/java/org/generationcp/middleware/api/germplasm/GermplasmGuidGenerator.java

```diff
@@ -0,0 +1,35 @@
+package org.generationcp.middleware.api.germplasm;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.generationcp.middleware.pojos.Germplasm;
+import org.generationcp.middleware.pojos.workbench.CropType;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+import java.util.UUID;
+
+public class GermplasmGuidGenerator {
+
+	public static final String MID_STRING = "G";
+	public static final int SUFFIX_LENGTH = 15;
+
+	public static void generateGermplasmGuids(final CropType crop, final List<Germplasm> germplasmList) {
+		Preconditions.checkNotNull(crop);
```

Comment:
This looks interesting. I wonder if it could be abstracted even more. Something like this:
```java
// maybe in org.generationcp.middleware.util.Util?
public static <T extends UIDAssignable> void generateUIDs(
	final CropType crop, final List<T> list,
	final String midString,
	final Object suffixLength) {

	final boolean doUseUUID = crop.isUseUUID();
	for (final T item : list) {
		if (item.getUID() == null) {
			if (doUseUUID) {
				item.setUID(UUID.randomUUID().toString());
			} else {
				final String cropPrefix = crop.getPlotCodePrefix();
				item.setUID(cropPrefix + midString + randomAlphanumeric(suffixLength));
			}
		}
	}
}
...
private static interface UIDAssignable {
	void setUID(String uid);
	String getUID();
}

```

Reply:
I have create https://ibplatform.atlassian.net/browse/IBP-4167